auth/wrapper.go

```diff
@@ -119,6 +119,8 @@ func (a *AuthorizerWrapper) Token() (*oauth2.Token, error) {
 	var adalToken adal.Token
 	if spToken, ok := tokenProviders[0].(ServicePrincipalToken); ok && spToken != nil {
 		adalToken = spToken.Token()
+	} else if spToken, ok := tokenProviders[0].(*adal.Token); ok && spToken != nil {
+		adalToken = *spToken
```

Comment:
Just renaming this variable to make it clearer
```suggestion
	} else if token, ok := tokenProviders[0].(*adal.Token); ok && token != nil {
		adalToken = *token
```